Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs

```diff
@@ -839,35 +840,32 @@ eval (TimeUpdate time lat lng) state = do
-      true, ST.RideAccepted, false, Just snapped -> do
-
-        let dist = (getDistanceBwCordinates driverLat driverLong snapped.lat snapped.lon)
-            isDriverNearBy = ( dist < state.data.config.waitTimeConfig.thresholdDist)
-        pure if isDriverNearBy then NotifyAPI else AfterRender
-      _, _, _, _ -> pure AfterRender
+    case state.data.config.waitTimeConfig.enableWaitTime, state.props.currentStage, state.data.activeRide.notifiedCustomer, state.data.latLonArray of
+      true, ST.RideAccepted, false, Just lastLoc -> do
+        let lastTwoHB = checkTwoHB {lat : driverLat, lon:driverLong, place: ""} {lat:lastLoc.lat, lon : lastLoc.lon, place : "" } {lat:state.data.activeRide.src_lat, lon : state.data.activeRide.src_lon, place : "" } state.data.config.waitTimeConfig.thresholdDist
+        pure if lastTwoHB then UpdateAndNotify {lat:state.data.activeRide.src_lat, lon : state.data.activeRide.src_lon, place : "" } lastTwoHB else (UpdateLastLoc driverLat driverLong)
+      _, _, _, _-> pure $ UpdateLastLoc driverLat driverLong
     ]
 
-eval (UpdateOriginDist lat lng) state =
-  continueWithCmd state [ do
-    void $ launchAff $ flowRunner defaultGlobalState $ do
-      push <- liftFlow $ getPushFn Nothing "HomeScreen"
-      rideRouteResp <- Remote.rideRoute state.data.activeRide.id
-      case rideRouteResp of
-        Left _ -> pure unit
-        Right (API.RideRouteResp resp) -> 
-          case Array.head resp.points of
-            Just (API.LatLong loc) -> do
-              let dist = getDistanceBwCordinates lat lng loc.lat loc.lon
-              liftFlow $ push $ UpdateAndNotify {lat : loc.lat, lon : loc.lon, place : ""} $ dist < state.data.config.waitTimeConfig.thresholdDist
-              pure unit
-            _ -> pure unit
-      pure unit
-    pure NoAction]
+eval (UpdateLastLoc lat lon) state = continue state {data { latLonArray = Just {lat : lat, lon : lon, place : ""}}}
   
-eval (UpdateAndNotify snappedOrigin callNotify) state = do
-  continueWithCmd state{data{snappedOrigin = Just snappedOrigin}} [ pure if callNotify then NotifyAPI else NoAction]
+eval (UpdateAndNotify snappedOrigin callNotify) state =
+  continueWithCmd state{data{snappedOrigin = Just snappedOrigin}}
+        [ do
+            void $ launchAff $ EHC.flowRunner defaultGlobalState $ runExceptT $ runBackT
+              $ do
+                  push <- liftFlowBT $ getPushFn Nothing "HomeScreen"
+                  if callNotify then do
+                        GetRouteResp routeApiResponse <- Remote.getRouteBT (Remote.makeGetRouteReq state.data.currentDriverLat state.data.currentDriverLon snappedOrigin.lat snappedOrigin.lon) "pickup"
+                        let shortRoute = (routeApiResponse Array.!! 0)
+                        liftFlowBT $ push $ case shortRoute of
+                            Just (Route route) -> do
+                                if route.distance <= 30 then NotifyAPI else NoAction
```

Comment:
Remove this hardcoded value and add config
And instead of `NoAction` call `UpdateLastLoc`